c/driver/bigquery/bigquery_test.cc

```diff
@@ -0,0 +1,359 @@
+
+ public:
+  // will need to be updated to SetUpTestSuite when gtest is upgraded
+  static void SetUpTestCase() {
+    if (quirks_.skip_) {
+      GTEST_SKIP();
+    }
+
+    struct AdbcError error;
+    struct AdbcDatabase db;
+    struct AdbcConnection connection;
+    struct AdbcStatement statement;
+
+    std::memset(&error, 0, sizeof(error));
+    std::memset(&db, 0, sizeof(db));
+    std::memset(&connection, 0, sizeof(connection));
+    std::memset(&statement, 0, sizeof(statement));
+
+    ASSERT_THAT(AdbcDatabaseNew(&db, &error), IsOkStatus(&error));
+    ASSERT_THAT(quirks_.SetupDatabase(&db, &error), IsOkStatus(&error));
+    ASSERT_THAT(AdbcDatabaseInit(&db, &error), IsOkStatus(&error));
+
+    ASSERT_THAT(AdbcConnectionNew(&connection, &error), IsOkStatus(&error));
+    ASSERT_THAT(AdbcConnectionInit(&connection, &db, &error), IsOkStatus(&error));
+
+    std::string schema_name = "ADBC_TESTING_" + GetUuid();
+    std::string query = "CREATE SCHEMA `";
+    query += quirks_.catalog_name_;
+    query += "." + schema_name + "`";
+
+    ASSERT_THAT(AdbcStatementNew(&connection, &statement, &error), IsOkStatus(&error));
+    ASSERT_THAT(AdbcStatementSetSqlQuery(&statement, query.c_str(), &error),
+                IsOkStatus(&error));
+    ASSERT_THAT(AdbcStatementExecuteQuery(&statement, nullptr, nullptr, &error),
+                IsOkStatus(&error));
+
+    quirks_.schema_ = schema_name;
+
+    ASSERT_THAT(AdbcStatementRelease(&statement, &error), IsOkStatus(&error));
+    ASSERT_THAT(AdbcConnectionRelease(&connection, &error), IsOkStatus(&error));
+    ASSERT_THAT(AdbcDatabaseRelease(&db, &error), IsOkStatus(&error));
+  }
+
+  // will need to be updated to TearDownTestSuite when gtest is upgraded
+  static void TearDownTestCase() {
+    if (quirks_.skip_) {
+      GTEST_SKIP();
+    }
+
+    struct AdbcError error;
+    struct AdbcDatabase db;
+    struct AdbcConnection connection;
+    struct AdbcStatement statement;
+
+    std::memset(&error, 0, sizeof(error));
+    std::memset(&db, 0, sizeof(db));
+    std::memset(&connection, 0, sizeof(connection));
+    std::memset(&statement, 0, sizeof(statement));
+
+    ASSERT_THAT(AdbcDatabaseNew(&db, &error), IsOkStatus(&error));
+    ASSERT_THAT(quirks_.SetupDatabase(&db, &error), IsOkStatus(&error));
+    ASSERT_THAT(AdbcDatabaseInit(&db, &error), IsOkStatus(&error));
+
+    ASSERT_THAT(AdbcConnectionNew(&connection, &error), IsOkStatus(&error));
+    ASSERT_THAT(AdbcConnectionInit(&connection, &db, &error), IsOkStatus(&error));
+
+    std::string query = "DROP SCHEMA `ADBC_TESTING." + quirks_.schema_ + "`";
+
+    ASSERT_THAT(AdbcStatementNew(&connection, &statement, &error), IsOkStatus(&error));
+    ASSERT_THAT(AdbcStatementSetSqlQuery(&statement, query.c_str(), &error),
+                IsOkStatus(&error));
+    ASSERT_THAT(AdbcStatementExecuteQuery(&statement, nullptr, nullptr, &error),
+                IsOkStatus(&error));
+
+    quirks_.schema_ = "ADBC_TESTING";
```

Comment:
instead of using `ADBC_TESTING` directly, should use the values in the quirks objects.